backend/common/src/main/java/ai/verta/modeldb/common/subtypes/CommonArtifactHandler.java

```diff
@@ -130,18 +134,17 @@ protected abstract Query buildGetArtifactsQuery(
         uploadCompleted = true;
       }
 
-      var validPrefix = artifactStoreConfig.getPathPrefixWithSeparator() + this.entityName;
-      var path = validPrefix + "/" + entityId + "/" + artifact.getKey();
-      var storeTypePath =
-          !artifact.getPathOnly() ? artifactStoreConfig.storeTypePathPrefix() + path : "";
+      var path = artifact.getPath();
+      if (!isPopulateFromOtherEntity) {
+        var validPrefix = artifactStoreConfig.getPathPrefixWithSeparator() + this.entityName;
+        path = validPrefix + "/" + entityId + "/" + artifact.getKey();
+        artifact = artifact.toBuilder().setPath(path).build();
+      }
+      var storeTypePath = !artifact.getPathOnly() ? path : "";
```

Comment:
`storeTypePath` here has a different value from before, so it has to be checked with care (we can't rely on previously working tests)

Reply:
no, it has same value. Before changes we are setting it if path_only is false other wise we will leave it empty. so here after changes we keep it as it is. it is not populate in artifact proto. it is populate in DB only.